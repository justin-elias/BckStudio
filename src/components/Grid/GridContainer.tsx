// Core
import React from "react";
import {Grid, GridProps} from "@mui/material";
// Style
//@ts-ignore
import classNames from "classnames";
// Assets

export default function GridContainer(props: GridProps) {
    const { children, className, ...rest } = props;

  return (
    <Grid container {...rest} className={className}>
      {children}
    </Grid>
  );
};