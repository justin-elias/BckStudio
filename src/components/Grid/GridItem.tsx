// Core
import React from "react";
// Style
import {makeStyles} from "@mui/styles";
//@ts-ignore
import classNames from "classnames";
// Components
import {Grid, GridProps} from "@mui/material";
// Assets

/* eslint-disable no-unused-vars */
const useStyles = makeStyles(() => ({
    /* eslint-enable no-unused-vars */

    grid: {
    position: "relative",
    width: "100%",
    minHeight: "1px",
    paddingRight: "15px",
    paddingLeft: "15px",
    flexBasis: "auto"
  }

}));

export default function GridItem(props: GridProps) {
  const classes = useStyles(props);
  const { children, className, ...rest } = props;
  return (
    <Grid item {...rest} className={classes.grid + " " + className}>
      {children}
    </Grid>
  );
};
