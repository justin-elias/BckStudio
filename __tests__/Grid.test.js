import React from "react";
import '@testing-library/jest-dom';
import { render, fireEvent } from "@testing-library/react";
import GridContainer from "../src/components/Grid/GridContainer";
import GridItem from "../src/components/Grid/GridItem";

describe('<GridContainer />', () => {
    it('should render correctly', () => {
        const {container} = render(<GridContainer/>);
        expect(container).toMatchSnapshot();
    });

    it('should render children', () => {
        const {container} = render(
            <GridContainer>
                <GridItem xs={12}>
                    <div>Test</div>
                </GridItem>
            </GridContainer>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render children with custom className', () => {
        const {container} = render(
            <GridContainer className="custom-class">
                <GridItem xs={12}>
                    <div>Test</div>
                </GridItem>
            </GridContainer>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render children with custom style', () => {
        const {container} = render(
            <GridContainer style={{backgroundColor: 'red'}}>
                <GridItem xs={12}>
                    <div>Test</div>
                </GridItem>
            </GridContainer>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render children with custom className and style', () => {
        const {container} = render(
            <GridContainer className="custom-class" style={{backgroundColor: 'red'}}>
                <GridItem xs={12}>
                    <div>Test</div>
                </GridItem>
            </GridContainer>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render children with custom className and style', () => {
        const {container} = render(
            <GridContainer className="custom-class" style={{backgroundColor: 'red'}}>
                <GridItem xs={12}>
                    <div>Test</div>
                </GridItem>
            </GridContainer>
        );
        expect(container).toMatchSnapshot();
    });


});
