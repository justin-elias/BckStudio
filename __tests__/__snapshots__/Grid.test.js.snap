// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContainer /> should render children 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-grid-1 undefined css-1idn90j-MuiGrid-root"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContainer /> should render children with custom className 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container custom-class css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-grid-2 undefined css-1idn90j-MuiGrid-root"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContainer /> should render children with custom className and style 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container custom-class css-11lq3yg-MuiGrid-root"
    style="background-color: red;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-grid-4 undefined css-1idn90j-MuiGrid-root"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContainer /> should render children with custom className and style 2`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container custom-class css-11lq3yg-MuiGrid-root"
    style="background-color: red;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-grid-5 undefined css-1idn90j-MuiGrid-root"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContainer /> should render children with custom style 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    style="background-color: red;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-grid-3 undefined css-1idn90j-MuiGrid-root"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContainer /> should render correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  />
</div>
`;
